cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsLaunchService.java

```diff
@@ -207,6 +208,21 @@ private ModelContext buildDefaultModelContext(AuthenticatedContext ac, CloudStac
             .withPrefixListIds(getPrefixListIds(amazonEC2Client, regionName, network.getOutboundInternetTraffic()))
             .withEncryptedAMIByGroupName(encryptedImageCopyService.createEncryptedImages(ac, stack, resourceNotifier));
 
+        AwsEfsFileSystem efsFileSystem = null;
+
+        // TODO: S3 uses this field. Need to add another field in CloudStack for EFS.
+        // Ideally, use a list of SpiFileSystem
+        if (stack.getFileSystem().isPresent()) {
```

Comment:
Why don't you combine this and the if statement below? So if this is true, `modelContext.withEnableEfs(true);`, etc., and if not `modelContext.withEnableEfs(false);`

Reply:
this part is not final. I have inserted testing code to fake EFS request and test. It was removed in this code review. The official way is to add EFS request from Cloud API, then we can extract it here. I am making changes to remove the TODO here as well.